src/adapters/discord.js

```diff
@@ -0,0 +1,87 @@
+const Discord = require('discord.js');
+const debug = require('debug')('bot:adapter:discord');
+
+const isString = require('lodash/isString');
+const isObject = require('lodash/isObject');
+const isArray = require('lodash/isArray');
+const isEmpty = require('lodash/isEmpty');
+
+const { discord: discordCfg } = require('../config');
+
+const discordAdapter = () => {};
+
+discordAdapter.__INIT__ = function ({ process }) {
+    const discordBot = new Discord.Client();
+
+    discordBot.on('ready', () => {
+        debug('Discrodjs ready');
+    });
+
+    discordBot.on('message', (msg) => {
+        // anti-bot + anti-self-loop
+        if (msg.author.bot) {
+            return;
+        }
+
+        const handle = (context) => {
+            const { output, reactions } = context;
+
+            // TODO rework!
+            if (!isEmpty(reactions)) {
+                reactions.forEach((reaction) => {
+                    msg
+                        .react(reaction)
```

Comment:
Don't we want to await here?
And where we `send`
I know, it is async and stuff, but it can look confusing sometimes